apis/run/v1alpha3/kappcontrollerconfig_types.go

```diff
@@ -0,0 +1,127 @@
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=tanzu-package-repo-global
+	GlobalNamespace string `json:"globalNamespace,omitempty"`
+
+	Deployment KappDeployment `json:"deployment,omitempty"`
+
+	Config KappConfig `json:"config,omitempty"`
+}
+
+type KappDeployment struct {
+	// Whether to enable host networking for kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+
+	// The priority value that various system components use to find the priority of the kapp-controller pod
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=tanzu-package-repo-global
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+
+	// Concurrency of kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=4
+	Concurrency int `json:"concurrency,omitempty"`
+
+	// kapp-controller deployment tolerations
+	// +kubebuilder:validation:Optional
+	Tolerations []map[string]string `json:"tolerations,omitempty"`
+
+	// Bind port for kapp-controller API
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=10350
+	ApiPort int `json:"apiPort,omitempty"`
+
+	// Address for metrics server
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=:8080
+	MetricsBindAddress string `json:"metricsBindAddress,omitempty"`
+}
+
+type KappConfig struct {
+	// A cert chain of trusted CA certs. These will be added to the system-wide cert pool of trusted CA's
+	// +kubebuilder:validation:Optional
+	CaCerts string `json:"caCerts,omitempty"`
+
+	// The url/ip of a proxy for kapp controller to use when making network requests
+	// +kubebuilder:validation:Optional
+	HttpProxy string `json:"httpProxy,omitempty"`
+
+	// The url/ip of a TLS capable proxy for kapp-controller to use when making network requests
+	// +kubebuilder:validation:Optional
+	HttpsProxy string `json:"httpsProxy,omitempty"`
+
+	// A comma delimited list of domain names which kapp-controller should bypass the proxy for when making requests
+	// +kubebuilder:validation:Optional
+	NoProxy string `json:"noProxy,omitempty"`
+
+	// A comma delimited list of hostnames for which kapp-controller should skip TLS verification
+	// +kubebuilder:validation:Optional
+	DangerousSkipTLSVerify string `json:"dangerousSkipTLSVerify,omitempty"`
+}
+
+// KappControllerConfigStatus defines the observed state of KappControllerConfig
+type KappControllerConfigStatus struct {
+	// Reference to the data value secret created by controller
+	SecretRef SecretRef `json:"secretRef,omitempty"`
+}
+
+type SecretRef struct {
+	// Name of the data value secret created by controller
+	// +kubebuilder:validation:Optional
+	Name string `json:"name,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
```

Comment:
We should add kubebuilder:printcolumn for required params.

Reply:
Added `namespace`, `globalNamespace` and 'secretRef' as printable.